cmd/cli/user.go

```diff
@@ -49,10 +59,81 @@ func deleteUser(database moira.Database, user string) error {
 	if err != nil {
 		return err
 	}
+
 	for _, contactID := range contactIDs {
 		if err := database.RemoveContact(contactID); err != nil {
 			return err
 		}
 	}
+
 	return nil
 }
+
+func handleCleanup(logger moira.Logger, database moira.Database, config cleanupConfig) error {
+	var users []string
+
+	reader := json.NewDecoder(bufio.NewReader(os.Stdin))
+
+	if err := reader.Decode(&users); err != nil {
+		return err
+	}
+
+	return usersCleanup(logger, database, users, config)
+}
+
+func usersCleanup(logger moira.Logger, database moira.Database, users []string, config cleanupConfig) error {
+	usersMap := make(map[string]bool, len(users)+len(config.Whitelist))
+
+	if config.HandlingAnonymous {
+		config.Whitelist = append(config.Whitelist, "")
+	}
+
+	for _, user := range append(users, config.Whitelist...) {
+		usersMap[user] = true
+	}
+
+	contacts, err := database.GetAllContacts()
+	if err != nil {
+		return err
+	}
+
+	var usersNotFound []string
+
+	for _, contact := range contacts {
+		if !usersMap[contact.User] {
+			usersNotFound = append(usersNotFound, contact.User)
+		}
+	}
+
+	for _, user := range usersNotFound {
+		if config.Delete {
+			if err = deleteUser(database, user); err != nil {
```

Comment:
It's a bit confusing that we have things like ("disable and save user subscritions") going right after ("delete user")
even if it would never happen
can we make this part of code more clear?